src/Compilers/CSharp/Test/Emit3/OverloadResolutionPriorityTests.cs

```diff
@@ -0,0 +1,2005 @@
+                [OverloadResolutionPriority(1)]
+                public static C1 operator +(C1 c1, C2 c2) => throw null;
+            }
+
+            public class C2
+            {
+                public static C2 operator +(C1 c1, C2 c2) => throw null;
+            }
+            ";
+
+        var executable = @"
+            var c1 = new C1();
+            var c2 = new C2();
+            _ = c1 + c2;
+            ";
+
+        var expectedDiagnostics = new[] {
+            // (4,17): error CS0034: Operator '+' is ambiguous on operands of type 'C1' and 'C2'
+            //             _ = c1 + c2;
+            Diagnostic(ErrorCode.ERR_AmbigBinaryOps, "c1 + c2").WithArguments("+", "C1", "C2").WithLocation(4, 17)
+        };
+
+        CreateCompilation([executable, source, OverloadResolutionPriorityAttributeDefinition]).VerifyDiagnostics(expectedDiagnostics);
+
+        var comp = CreateCompilation([source, OverloadResolutionPriorityAttributeDefinition]);
+        CreateCompilation(executable, references: [AsReference(comp, useMetadataReference)]).VerifyDiagnostics(expectedDiagnostics);
+    }
+
+    [Theory, CombinatorialData]
+    public void UnaryOperator_Allowed(bool useMetadataReference)
+    {
+        var source = """
+            using System.Runtime.CompilerServices;
+
+            public struct S
+            {
+                public static S? operator!(S? s) => throw null;
+
+                [OverloadResolutionPriority(1)]
+                public static S operator!(S s)
+                {
+                    System.Console.Write("1");
+                    return s;
+                }
+            }
+            """;
+
+        var executable = """
+            S? s = new S();
+            _ = !s;
+            """;
+
+        CompileAndVerify([executable, source, OverloadResolutionPriorityAttributeDefinition], expectedOutput: "1").VerifyDiagnostics();
+
+        var comp = CreateCompilation([source, OverloadResolutionPriorityAttributeDefinition]);
+        CompileAndVerify(executable, references: [AsReference(comp, useMetadataReference)], expectedOutput: "1").VerifyDiagnostics();
+    }
+
+    [Theory]
+    [InlineData("implicit")]
+    [InlineData("explicit")]
+    public void ConversionOperators_Disallowed(string operatorType)
+    {
+        var source = $$"""
+            using System.Runtime.CompilerServices;
+
+            public class C
+            {
+                [OverloadResolutionPriority(1)]
+                public static {{operatorType}} operator C(int i) => throw null;
+            }
+            """;
+
+        CreateCompilation([source, OverloadResolutionPriorityAttributeDefinition]).VerifyDiagnostics(
+            // (5,6): error CS9501: Cannot use 'OverloadResolutionPriorityAttribute' on this member.
+            //     [OverloadResolutionPriority(1)]
+            Diagnostic(ErrorCode.ERR_CannotApplyOverloadResolutionPriorityToMember, "OverloadResolutionPriority(1)").WithLocation(5, 6)
+        );
+    }
+}
```

Comment:
Do we have a test for method group conversion?
Consider also adding a test where override (in metadata) has a different priority and is ignored. We can observe the value on PEMethodSymbol too

Reply:
Added the first. The second is `Overids_ChangePriorityInMetadata_Methods/Indexers` above.